### PR TITLE
Support passing a ssl context to run-jetty

### PR DESCRIPTION
**Pull request by mcorbin**
This commit adds a new option named :ssl-context to configure
certificates using a Java SSLContext instead of a keystore file.
It allows users to easily configure jetty with SSL from certs files,
by using the less-awful-ssl library to create the SSLContext for
example (as shown in the tests).

### CODE REVIEW

**Comment by weavejester** [comment]
Thanks for the PR. Please ensure it adheres to the [contributing guidelines](https://github.com/ring-clojure/ring/blob/master/CONTRIBUTING.md).

**Comment by mcorbin** [comment]
> Thanks for the PR. Please ensure it adheres to the contributing guidelines.

I updated the commit.

**Comment by weavejester** [comment]
Can you change the commit message to:

```
Add option for passing an SSLContext to Jetty

Adds a new option named :ssl-context to configure certificates using a
SSLContext instance instead of a keystore file. This allows interop with
libraries like less-awful-ssl.
```

File: ring-jetty-adapter/src/ring/adapter/jetty.clj [hunk 1]
```diff
@@ -146,6 +148,7 @@
   :ssl?                 - allow connections over HTTPS
   :ssl-port             - the SSL port to listen on (defaults to 443, implies
                           :ssl? is true)
+  :ssl-context          - the SSLContext to use for SSL connections
```
Comment (weavejester) [inline ring-jetty-adapter/src/ring/adapter/jetty.clj:151]: As it's optional, this should be changed to:

```
  :ssl-context          - an optional SSLContext to use for SSL connections
```

Comment (mcorbin) [reply ring-jetty-adapter/src/ring/adapter/jetty.clj]: I've updated the description, thanks.

**Comment by mcorbin** [comment]
> Can you change the commit message to:

Thanks you, I've updated the commit message.

File: ring-jetty-adapter/test/ring/adapter/test/jetty.clj [hunk 2]
```diff
@@ -35,6 +36,12 @@
 (defn- all-threads []
   (.keySet (Thread/getAllStackTraces)))
 
+(defn ssl-context
+  []
```
Comment (weavejester) [inline ring-jetty-adapter/test/ring/adapter/test/jetty.clj:40]: As a small point, could you put the `[]` on the previous line?

Comment (mcorbin) [reply ring-jetty-adapter/test/ring/adapter/test/jetty.clj]: done ;)

**Pull request merged**